src/modules/water-demand-prediction/water-demand-prediction.component.ts

```diff
@@ -0,0 +1,545 @@
+import {DropdownComponent} from "../../common/components/dropdown/dropdown.component";
+import {stringToColor} from "../../common/stringToColor";
+
+import "dayjs/locale/de";
+import "dayjs/locale/en";
+
+dayjs.extend(duration, relativeTime);
+
+type MeterNames = WaterDemandPredictionService.MeterNames;
+type WeatherColumns = WaterDemandPredictionService.WeatherColumns;
+type SingleSmartmeter = WaterDemandPredictionService.SingleSmartmeter;
+type PredictedSmartmeter = WaterDemandPredictionService.PredictedSmartmeter;
+
+@Component({
+  selector: "water-demand-prediction",
+  imports: [BaseChartDirective, CommonModule, DropdownComponent, TranslatePipe],
+  templateUrl: "./water-demand-prediction.component.html",
+})
+export class WaterDemandPredictionComponent {
+  private waterDemandService = inject(WaterDemandPredictionService);
+
+  protected minHeightCharts: string = "15em";
+  protected standardHeightCharts: string = "50%";
+
+  /** the displayed resolution in the charts of real data */
+  protected displayedResolution = signal<string | undefined>("hourly");
+
+  private currentLang = "water-demand-prediction.dayjs.locale";
+  //BUG: dayjs sets locale when initializing variables, cant dynamically chagne them
+
+  /** variables timeframe dropdown */
+  protected menuTime = "water-demand-prediction.choice.timeframe";
+  protected optionsTime: Record<string, string> = {
+    "one day": dayjs.duration(1, "days").humanize(),
+    "one week": dayjs.duration(1, "week").humanize(),
+    "one month": dayjs.duration(1, "month").humanize(),
+    "three months": dayjs.duration(3, "month").humanize(),
+    "six months": dayjs.duration(6, "month").humanize(),
+    "one year": dayjs.duration(1, "year").humanize(),
+    all: dayjs.duration(3, "year").humanize(),
+  };
+  readonly choiceTime = signal<string | undefined>(undefined);
+
+  /** variables startpoint dropdown */
+  protected menuStartPoint = "water-demand-prediction.startpoint.menu";
+  private startOfData = dayjs(new Date(2021, 4, 26, 0, 0, 0)).format(
+    "YYYY-MM-DD HH:mm:ss",
+  );
+  private startofJune = dayjs(new Date(2021, 5, 1, 0, 0, 0)).format(
+    "YYYY-MM-DD HH:mm:ss",
+  );
+  private startOfYear22 = dayjs(new Date(2022, 0, 1, 0, 0, 0)).format(
+    "YYYY-MM-DD HH:mm:ss",
+  );
+  protected optionsStartPoint: Record<string, string> = {
+    [this.startOfData]: "water-demand-prediction.startpoint.options.a",
+    [this.startofJune]: "water-demand-prediction.startpoint.options.b",
+    [this.startOfYear22]: "water-demand-prediction.startpoint.options.c",
+  };
+  readonly choiceStartPoint = signal<string | undefined>(undefined);
+
+  /** variables resolution dropdown */
+  protected menuResolution = "water-demand-prediction.choice.resolution";
+  protected optionsResolution: Record<string, string> = {
+    hourly: "water-demand-prediction.resolution.hourly",
+    daily: "water-demand-prediction.resolution.daily",
+    weekly: "water-demand-prediction.resolution.weekly",
+  };
```

Comment:
This is just a prefix plus the key, translation keys like this can easily be constructed in the html.